flocker/node/test/test_script.py

```diff
@@ -226,6 +230,21 @@ def test_initialized(self):
             ),
         )
 
+    def test_initialized_hostname(self):
+        host = b"192.0.2.13"
+        port = 2314
+        name = u"from_config-test"
+
+        setup_config(self, control_address=host, control_port=port, name=name)
+        options = DatasetAgentOptions()
+        options.parseOptions([b"--agent-config", self.config.path])
+        config = get_configuration(options)
+        config['hostname'] = 'hostname.example'
```

Comment:
So.... everywhere else we use IPs. This changes means we're OK with DNS hostnames appearing in e.g. REST API output? That might violate our schema.